From f24c1864cb12d1f63657b04a0feda410378cbfd1 Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Thu, 14 Mar 2024 08:35:50 +0200
Subject: [PATCH] [SW-165981] emb cmn: renamed i2csi to oob

ported the oob changes from preboot rtos to support the
emb common mods.

Change-Id: Ic47298fbc907bd9935d86c07c481e27b4cf0caf7
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/include/i2cslave_indirect.h b/include/i2cslave_indirect.h
index 27124a2..b6c178a 100644
--- a/include/i2cslave_indirect.h
+++ b/include/i2cslave_indirect.h
@@ -10,6 +10,7 @@
 
 #include <hl_fw_if.h>
 #include <rma_if.h>
+#include <oob_if.h>
 
 u8 i2csi_req_handle_sts(u8 cmd);
 u8 i2csi_req_handle_rdata(u8 cmd, u32 read_seq);
@@ -23,9 +24,6 @@
 bool i2csi_req_is_process_op(void);
 bool i2csi_req_op_is_fast(void);
 bool i2csi_ctrl_is_rst_dev(void);
-void i2csi_set_boot_err(enum hl_fw_fit_boot_error err,
-			bool is_linux_boot_fail);
-int i2csi_set_wd_err(enum hl_fw_reset_cause err);
 void i2csi_update_smbus_alert(struct rma_entry *rma_header);
 void i2csi_set_rma_failure(struct rma_failure *rma_f);
 #endif /* _I2CSLAVE_INDIRECT_H_ */
diff --git a/include/oob_if.h b/include/oob_if.h
new file mode 100644
index 0000000..b0e710b
--- /dev/null
+++ b/include/oob_if.h
@@ -0,0 +1,17 @@
+/* SPDX-License-Identifier:     Apache-2.0 */
+/*
+ * Copyright (C) 2017-2024 HabanaLabs Ltd.
+ * All Rights Reserved.
+ *
+ */
+
+#ifndef _OOB_IF_H_
+#define _OOB_IF_H_
+
+#include <hl_fw_if.h>
+
+void oob_set_boot_err(enum hl_fw_fit_boot_error err,
+			bool is_linux_boot_fail);
+int oob_set_wd_err(enum hl_fw_reset_cause err);
+
+#endif /* _OOB_IF_H_ */
diff --git a/lib/preboot/fit.c b/lib/preboot/fit.c
index 46fea22..6613444 100644
--- a/lib/preboot/fit.c
+++ b/lib/preboot/fit.c
@@ -940,7 +940,7 @@
 		break;
 	};
 
-	i2csi_set_boot_err(boot_err, false);
+	oob_set_boot_err(boot_err, false);
 }
 
 static int fwupd_sec_convert_min_error(enum fit_sec_failure err)
diff --git a/lib/preboot/i2cslave_indirect.c b/lib/preboot/i2cslave_indirect.c
index 84bdb85..f7d6319 100644
--- a/lib/preboot/i2cslave_indirect.c
+++ b/lib/preboot/i2cslave_indirect.c
@@ -1108,7 +1108,7 @@
 		smbus_alert.fatal_error |= HL_I2CI_SMBUS_FATAL_ERROR_HBM_INIT;
 }
 
-void i2csi_set_boot_err(enum hl_fw_fit_boot_error err, bool is_linux_boot_fail)
+void oob_set_boot_err(enum hl_fw_fit_boot_error err, bool is_linux_boot_fail)
 {
 	struct hl_i2ci_error_status_fit_failure *fit_fail;
 	struct hl_i2ci_agent_signature_failure *sig_fail;
@@ -1168,7 +1168,7 @@
 }
 
 /* running from module_lkd fast context - please consider when changing */
-int i2csi_set_wd_err(enum hl_fw_reset_cause err)
+int oob_set_wd_err(enum hl_fw_reset_cause err)
 {
 	u32 err_sts_wd_err = HL_ERROR_STATUS_OTHER |
 					(HL_ERROR_STATUS_OTHER_WATCHDOG << 8);
diff --git a/lib/preboot/module_lkd.c b/lib/preboot/module_lkd.c
index 961301f..feb6b7a 100644
--- a/lib/preboot/module_lkd.c
+++ b/lib/preboot/module_lkd.c
@@ -87,10 +87,10 @@
 
 	switch (mcomms->reset_cause) {
 	case HL_RESET_CAUSE_HEARTBEAT:
-		err = i2csi_set_wd_err(HL_FW_RESET_CAUSE_HEARTBEAT);
+		err = oob_set_wd_err(HL_FW_RESET_CAUSE_HEARTBEAT);
 		break;
 	case HL_RESET_CAUSE_TDR:
-		err = i2csi_set_wd_err(HL_FW_RESET_CAUSE_TDR);
+		err = oob_set_wd_err(HL_FW_RESET_CAUSE_TDR);
 		/* no break */
 	default:
 		break;
